From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Mon, 14 Mar 2022 18:07:45 +1000
Subject: [PATCH] Use view distance for player mob caps

This is a uniquely MultiPaper issue as MultiPaper doesn't tick
chunks at all outside of the simulation distance.

diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index cc8fd59fbb3d039a303205bc6fae861cef9ad471..e2e2ff7bcebe35f1ef66b66ba445aff7ed5317e2 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -246,7 +246,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         this.playerGeneralAreaMap.add(player, chunkX, chunkZ, GENERAL_AREA_MAP_SQUARE_RADIUS); // Paper - optimise checkDespawn
         // Paper start - per player mob spawning
         if (this.playerMobDistanceMap != null) {
-            this.playerMobDistanceMap.add(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance());
+            this.playerMobDistanceMap.add(player, chunkX, chunkZ, this.viewDistance); // MultiPaper - use view distance instead of simulation distance
         }
         // Paper end - per player mob spawning
     }
@@ -286,7 +286,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         this.playerGeneralAreaMap.update(player, chunkX, chunkZ, GENERAL_AREA_MAP_SQUARE_RADIUS); // Paper - optimise checkDespawn
         // Paper start - per player mob spawning
         if (this.playerMobDistanceMap != null) {
-            this.playerMobDistanceMap.update(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance());
+            this.playerMobDistanceMap.update(player, chunkX, chunkZ, this.viewDistance); // MultiPaper - use view distance instead of simulation distance
         }
         // Paper end - per player mob spawning
     }
